### ISSUE TITLE
label constraints syntax error while `FOLLOWER_CONSTRAINT` use dict format

### ISSUE THREAD

**Issue by aytrack** [pingcap/tidb#29908]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. deploy a cluser with 4 tikv, 2 with label `{"disk": "ssd"}` and 2 with label `{"disk":"hdd"}` 
```sql
create placement policy ssd4 constraints="[+disk=ssd]" FOLLOWER_CONSTRAINTS="{+disk=ssd:1,+disk=hdd:2}" followers=3;
```
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)
create placement policy successfully.
### 3. What did you see instead (Required)
```sql
mysql> create placement policy ssd4 constraints="[+disk=ssd]" FOLLOWER_CONSTRAINTS="{+disk=ssd:1,+disk=hdd:2}" followers=3;
ERROR 1105 (HY000): label constraints in map syntax have invalid replicas: count of labels '+disk=ssd:1' should be positive, but got 0: invalid FollowerConstraints
```
### 4. What is your TiDB version? (Required)
```sql
mysql> select tidb_version();
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                                                                       |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v5.3.0
Edition: Community
Git Commit Hash: c83f032b04a40b3148dff712392dd924ae064132
Git Branch: heads/refs/tags/v5.3.0
UTC Build Time: 2021-11-16 11:55:42
GoVersion: go1.16.4
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```
<!-- Paste the output of SELECT tidb_version() -->

**Comment by xhebox** [comment]
This is a known limitation of the current parser. It could be workarounded by writing `FOLLOWER_CONSTRAINTS='{"+disk=ssd":1,"+disk=hdd":2};` or `FOLLOWER_CONSTRAINTS="{+disk=ssd: 1, +disk=hdd: 2}"`. 

Thus I don't think it is a major bug, more like a moderate one.

@morgo  PTAL, may be we should mention this in the document. This is the second time I saw this.

**Comment by xhebox** [comment]
Closed for #7024
